tools/llm/run_vlm.py

```diff
@@ -0,0 +1,387 @@
+    """
+
+    def __init__(self, lm):
+        super().__init__()
+        self.lm = lm
+
+    def forward(self, inputs_embeds, position_ids):
+        out = self.lm(inputs_embeds=inputs_embeds, position_ids=position_ids)
+        return out.logits if hasattr(out, "logits") else out
+
+
+def _compile_eagle2_lm(
+    language_model: torch.nn.Module,
+    input_embeds: torch.Tensor,
+    args: argparse.Namespace,
+) -> torch.nn.Module:
+    """
+    Compile Eagle2 language model with Torch-TensorRT.
+
+    The function follows the same precision-specific flag logic used in
+    *run_llm.py* for consistency.
+    """
+    lm_wrap = _LMNoCache(language_model).to(DEVICE).eval()
+    max_seq_len = input_embeds.shape[1] + args.num_tokens
+
+    S = torch.export.Dim("seq", min=1, max=max_seq_len)
+    position_ids = torch.arange(input_embeds.shape[1]).unsqueeze(0).to(DEVICE)
+    dyn_shapes = {"inputs_embeds": {1: S}, "position_ids": {1: S}}
+
+    # Precision-specific flags --------------------------------------------------#
+    use_fp32_acc = False
+    use_explicit_typing = False
+    if args.precision == "FP16":
+        enabled_precisions = {torch.float32}
+        use_fp32_acc = True
+        use_explicit_typing = True
+    elif args.precision == "BF16":
+        enabled_precisions = {torch.bfloat16}
+    else:  # FP32
+        enabled_precisions = {torch.float32}
+
+    with torch.inference_mode():
+        exported = torch.export.export(
+            lm_wrap,
+            (input_embeds, position_ids),
+            dynamic_shapes=dyn_shapes,
+            strict=False,
+        )
+
+    with torch_tensorrt.logging.debug() if args.debug else nullcontext():
+        trt_mod = torch_tensorrt.dynamo.compile(
+            exported,
+            inputs=[input_embeds, position_ids],
+            enabled_precisions=enabled_precisions,
+            use_explicit_typing=use_explicit_typing,
+            use_fp32_acc=use_fp32_acc,
+            device=DEVICE,
+            disable_tf32=True,
+            use_python_runtime=True,
+            debug=args.debug,
+            offload_module_to_cpu=True,
+            min_block_size=args.min_block_size,
+        )
+    return trt_mod
+
+
+def compile_torchtrt(
+    model: torch.nn.Module, args: argparse.Namespace
+) -> torch.nn.Module:
+    """
+    Front-end dispatcher mirroring *run_llm.py*’s `compile_torchtrt`.
+
+    Depending on the target VLM, delegates to the appropriate compile routine.
+    """
+    torch_dtype = {
+        "FP16": torch.float16,
+        "BF16": torch.bfloat16,
+    }.get(args.precision, torch.float32)
+
+    example_embeds = torch.randn(
+        1,
+        2560,
+        model.language_model.config.hidden_size,
+        dtype=torch_dtype,
+        device=DEVICE,
+    )
+
+    if args.model.lower() == "eagle2":
+        return _compile_eagle2_lm(model.language_model, example_embeds, args)
+
+    msg = f"Unsupported model for compilation: {args.model}"
```

Comment:
Consider changing this message similar to the comment above (by including list of supported models)